### ISSUE TITLE
Implementation of Encoder

### ISSUE THREAD

**Issue by farisxiong** [richardzhuang0412/EmbedLLM#1]
Amazing work!

However, I found that the paper lacks detailed information regarding $\Phi_m$ and $\Phi_q$ of the encoder. Could you please elaborate on how this part is implemented? Is there a need for training to obtain these parameters? If so, could you provide some guidance on the training process?

I look forward to your response.

**Comment by richardzhuang0412** [comment]
Hi FarisXiong,

In our implementation of MF, $\phi_m$ and $\phi_q$ are simply an embedding layers (check the initialization of `nn.Embedding` layer) instead of a separate network. Does that clarify things up for you?

**Comment by farisxiong** [comment]
Thank you for your reply. I still have a few questions. Could you explain the training process? Additionally, do the different models mentioned in the appendix utilize distinct embedding layers?

**Comment by richardzhuang0412** [comment]
So for training MF our goal is to predict, given model $M$ and question $q$, whether $M$ can correctly answer $q$. Our data is a "correctness matrix" where the ij-th entry is a binary label represent whether model i correctly answer question j. To do so we have a model embedding layer that takes a model index and output an embedding, and a similar question embedding layer with just an extra projection layer that aligns the question embedding to a same dimension as model embedding. These layers can be seen as simply lookup tables that gives the model/question embedding from a given query index ID. So for your second question all models use the same embedding layer as the embedding layer is like a storage of embeddings rather than a network but it's just the stored embeddings are updated during training through backprop. The model IDs and question IDs can be found on huggingface under `model_order.csv` and `question_order.csv`. 

For instance let's say a data point is "Qwen1.5-7B-Chat (the model with ID 0 in our data) correctly answer question #1 (question with ID 0)". Then the network will first go through the two embedding layers to get the embedding at index 0 for both layers, perform an element-wise product, and send the product through a linear binary classifier which is simply optimized using BCE.

**Comment by farisxiong** [comment]
Thank you for your patience. I have no other questions.
